.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,49 @@
+{
+    "name": "mrc-dev",
+    "image": "mrc:dev",
+    "runArgs": [
+        "--gpus=all",
+        "--network=host",
+        "--cap-add=SYS_PTRACE",
+        "--cap-add=SYS_NICE",
+        "--security-opt",
+        "seccomp=unconfined"
+    ],
+    // the image build script will setup a user in the contaienr with the calling user's name
+    // thus it is safe to set remoteUser (the user inside the container) to be name of the caller
+    "remoteUser": "${localEnv:USER}",
+    // vscode will look up uid/gid of the remoteUser on the host machine and map that uid/gid into the container
+    "updateRemoteUserUID": true,
+    "workspaceFolder": "/work",
+    "workspaceMount": "source=${localWorkspaceFolder},target=/work,type=bind,consistency=cached",
+    "containerEnv": {
+        // the post_start script will look for these envs, if they are set, git will be configured
+        "VSCODE_CONTAINER_GIT_USER": "${localEnv:VSCODE_CONTAINER_GIT_USER}",
+        "VSCODE_CONTAINER_GIT_EMAIL": "${localEnv:VSCODE_CONTAINER_GIT_EMAIL}"
+    },
+    "overrideCommand": true, // infinite loop of sleeps
+    // this is the command that will build the docker image
+    "initializeCommand": "SKIP_RUN=yes ./scripts/devel.sh",
```

Comment:
This isn't what `initializeCommand` is for. We should build the container by passing a `Dockerfile` in the `build.dockerfile` field.

See: https://containers.dev/implementors/spec/#dockerfile-based

Reply:
i was trying to reuse as much of the ci scripts as possible. time to pivot.